### PR TITLE
syntax: misc extractions

### PR DESCRIPTION
**Pull request by Centril**
Part of https://github.com/rust-lang/rust/pull/65324.

r? @petrochenkov

### CODE REVIEW

File: src/libsyntax/parse/sess.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                id,
+                msg: msg.into(),
+                lint_id,
+            });
+        });
+    }
+
+    /// Extend an error with a suggestion to wrap an expression with parentheses to allow the
+    /// parser to continue parsing the following operation as part of the same expression.
+    pub fn expr_parentheses_needed(
```
Comment (petrochenkov) [inline src/libsyntax/parse/sess.rs]: I'm pretty sure this doesn't belong to `sess.rs` even if it was there previously.

**Comment by petrochenkov** [comment]
r=me after rebase and fixing tests

File: src/libsyntax/parse/sess.rs (hunk 1 above)
Comment (Centril) [reply src/libsyntax/parse/sess.rs]: Yeah I agree, but I'm not quite sure where to put it tho and we can always move it later. =)

**Pull request merged**